### PR TITLE
Burning SP for inactive users

### PR DESCRIPTION
**Pull request by Andrei-Masilevich**

### CODE REVIEW

File: libraries/chain/database/block_tasks/process_account_registration_bonus_expiration.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+{
+    registration_pool_service_i& registration_pool_service = ctx.services().registration_pool_service();
+    account_service_i& account_service = ctx.services().account_service();
+    dynamic_global_property_service_i& dgp_service = ctx.services().dynamic_global_property_service();
+
+    asset bonus = account.bonus;
+
+    const account_object& account_obj = account_service.get_account(account.account);
+
+    if (account_obj.delegated_scorumpower.amount > 0)
```
Comment (abolinbsu) [inline libraries/chain/database/block_tasks/process_account_registration_bonus_expiration.cpp]: check scorumpower - delegated_scorumpower < bonus

File: libraries/chain/database/block_tasks/process_account_registration_bonus_expiration.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        wlog("Account '${a}' has delegated scorumpower ${d}. Registration funds ${f} can't be returnd.",
+             ("a", account_obj.name)("d", account_obj.delegated_scorumpower)("f", bonus));
+    }
+    else
```
Comment (abolinbsu) [inline libraries/chain/database/block_tasks/process_account_registration_bonus_expiration.cpp]: remove else, the code should execute anyway

File: tests/chain_tests/registration/burning_bonus.cpp [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    account_service_i& account_service;
+    registration_pool_service_i& registration_pool_service;
+
+    Actor alice;
+    Actor bob;
+};
+
+BOOST_FIXTURE_TEST_SUITE(registration_committee_burning_bonus, registration_committee_burning_bonus_fixture)
+
+SCORUM_TEST_CASE(burning_for_sigle_check)
```
Comment (alexplevako) [inline tests/chain_tests/registration/burning_bonus.cpp]: burning_for_sigle_check -> burn_sp_for_one_account

File: tests/chain_tests/registration/burning_bonus.cpp [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    generate_blocks(db.head_block_time() + SCORUM_EXPIRATON_FOR_REGISTRATION_BONUS.to_seconds());
+
+    BOOST_REQUIRE_EQUAL(account_service.get_account(alice.name).scorumpower, ASSET_NULL_SP);
+
+    BOOST_REQUIRE_EQUAL(registration_pool_service.get().balance, old_registration_pool_balance);
+
+    validate_database();
+}
+
+SCORUM_TEST_CASE(burning_for_multiple_check)
```
Comment (alexplevako) [inline tests/chain_tests/registration/burning_bonus.cpp]: burn_sp_for_two_accounts

File: tests/chain_tests/registration/burning_bonus.cpp [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+    BOOST_REQUIRE_EQUAL(account_service.get_account(alice.name).scorumpower, ASSET_NULL_SP);
+    BOOST_REQUIRE_GT(account_service.get_account(bob.name).scorumpower, ASSET_NULL_SP);
+
+    generate_blocks(db.head_block_time() + SCORUM_EXPIRATON_FOR_REGISTRATION_BONUS.to_seconds() / 2);
+
+    BOOST_REQUIRE_EQUAL(account_service.get_account(bob.name).scorumpower, ASSET_NULL_SP);
+}
+
+SCORUM_TEST_CASE(no_burning_for_active_user_check)
```
Comment (alexplevako) [inline tests/chain_tests/registration/burning_bonus.cpp]: dont_burn_for_active_user

**Pull request merged**